process/smartContract/hooks/blockChainHook.go

```diff
@@ -299,6 +299,7 @@ func (bh *BlockChainHookImpl) LastEpoch() uint32 {
 
 // GetStateRootHash returns the state root hash from the last committed block
 func (bh *BlockChainHookImpl) GetStateRootHash() []byte {
+	// TODO decide here, in this PR if we should use the block's root hash or the committed root hash from the chain handler
 	if !check.IfNil(bh.blockChain.GetCurrentBlockHeader()) {
```

Comment:
it is better to choose the commited hash.

Reply:
done